pilot/spec/build_manager_spec.rb

```diff
@@ -221,4 +223,61 @@
       end
     end
   end
+
+  describe "#upload" do
+    describe "spaceship login (via Manager.login)" do
+      let(:fake_build_manager) { Pilot::BuildManager.new }
+      let(:upload_options) do
+        {
+          apple_id: 'mock_apple_id',
+          skip_waiting_for_build_processing: true,
+          ipa: 'foo'
+        }
+      end
+
+      before(:each) do
+        allow(fake_build_manager).to receive(:fetch_app_platform).and_return('ios')
+
+        fake_ipauploadpackagebuilder = double
+        allow(fake_ipauploadpackagebuilder).to receive(:generate).and_return(true)
+        allow(FastlaneCore::IpaUploadPackageBuilder).to receive(:new).and_return(fake_ipauploadpackagebuilder)
+
+        fake_itunestransporter = double
+        allow(fake_itunestransporter).to receive(:upload).and_return(true)
+        allow(FastlaneCore::ItunesTransporter).to receive(:new).and_return(fake_itunestransporter)
+      end
+
+      it "is not executed when skip_waiting_for_build_processing and apple_id are set" do
+        fake_build_manager.upload(upload_options)
+      end
+
+      it "is executed when skip_waiting_for_build_processing and apple_id are not set" do
```

Comment:
There are no expectations in this spec. Perhaps change some of the `allow`s to `expect` ?

Reply:
Silly question: What's the difference?

Reply:
`expect` is similar to an `allow`, but it will fail if that method does _not_ get called by the end of the spec. Just by having `allow`, the spec will either throw an error or pass, but it cannot fail.

Reply:
Ah ok, then this really makes sense to change here. 
I built this test up from nothing until it let the code execute enough to pass.

Reply:
I added an appropriate `expect` to each of the two specs: Once that the `Manager.login` method (which triggers a spaceship login) is _NOT_ called, and once that it is called at least once.